api/NuageAPI_test.go

```diff
@@ -278,72 +282,83 @@ func TestVMCreateDelete(t *testing.T) {
-		t.Fatalf("Unable to resolve VM %s ", vmInfo["name"])
-	}
+	// // Comparing port's IP address on VRS and VSD
+	// if portIPOnVSD != portIPOnVRS {
+	// 	t.Fatal("Port IPs on VSD and VRS do not match.")
+	// } else {
+	// 	t.Logf("Port IPs on VSD and VRS match.")
+	// }
 
-	portIPOnVRS := portState[port.StateKeyIPAddress]
-	t.Logf("VM %s got resolved with an IP address %s on VRS", vmInfo["name"], portIPOnVRS)
+	// Verifying if entity exists in OVSDB
+	entityExists, err := vrsConnection.CheckEntityExists(vmInfo["vmuuid"])
+	if err != nil {
+		t.Fatal("Unable to verify if entity exists in OVSDB")
+	}
 
-	// Comparing port's IP address on VRS and VSD
-	if portIPOnVSD != portIPOnVRS {
-		t.Fatal("Port IPs on VSD and VRS do not match.")
+	if entityExists {
+		t.Logf("VM %s with UUID %s exists in OVSDB", vmInfo["name"], vmInfo["vmuuid"])
 	} else {
-		t.Logf("Port IPs on VSD and VRS match.")
+		t.Fatalf("VM %s with UUID %s does not exist in OVSDB", vmInfo["name"], vmInfo["vmuuid"])
 	}
 
-    	// Verifying if entity exists in OVSDB
-    	entityExists, err := vrsConnection.CheckEntityExists(vmInfo["vmuuid"])
-    	if err != nil {
-        	t.Fatal("Unable to verify if entity exists in OVSDB")
-    	}
-
-    	if entityExists {
-        	t.Logf("VM %s with UUID %s exists in OVSDB", vmInfo["name"], vmInfo["vmuuid"])
-    	} else {
-        	t.Fatalf("VM %s with UUID %s does not exist in OVSDB", vmInfo["name"], vmInfo["vmuuid"])
-    	}
-
 	// Performing cleanup of port/entity on VRS
 	err = cleanup(vrsConnection, vmInfo)
 	if err != nil {
 		t.Fatal("Unable to delete port from OVSDB table")
 	}
 
-	t.Logf("Waiting for 300 seconds before verifying port gets removed from VRS")
-	time.Sleep(time.Duration(300) * time.Second)
+	// select {
+	// case portInfo = <-portInfoUpdateChan:
+	// 	fmt.Println("Received delete from OVSDB")
+	// 	fmt.Println("Got unregistered: ", portInfo.Registered)
+	// case <-ticker.C:
+	// 	fmt.Println("No IP received from OVSDB")
+	// }
+	// t.Logf("Waiting for 300 seconds before verifying port gets removed from VRS")
+	// time.Sleep(time.Duration(300) * time.Second)
 
 	// Verifying port deletion on VSD
-	if portDeletionFailure, vsdErr := util.VerifyVSDPortDeletion(Root, Enterprise, Domain, Zone, vmInfo["entityport"]); vsdErr != nil || portDeletionFailure {
-		t.Fatal("Deleted VM port still present on VSD")
-	}
+	// if portDeletionFailure, vsdErr := util.VerifyVSDPortDeletion(Root, Enterprise, Domain, Zone, vmInfo["entityport"]); vsdErr != nil || portDeletionFailure {
+	// 	t.Fatal("Deleted VM port still present on VSD")
+	// }
 
-	portState, _ = vrsConnection.GetPortState(vmInfo["entityport"])
+	portState, _ := vrsConnection.GetPortState(vmInfo["entityport"])
 
 	if _, ok := portState[port.StateKeyIPAddress]; ok {
 		t.Fatal("Entry for deleted VM Port still present in OVSDB table")
 	}
 
 	t.Logf("VM %s got removed from VRS successfully", vmInfo["name"])
+
+	vrsConnection.Disconnect()
```

Comment:
Remove all instances of these and use defer statement after connect. Makes code more legitimate to read.